test/scale-infra/deploy_ks_core.yaml

```diff
@@ -0,0 +1,110 @@
+             kubectl --context kscore delete ns open-cluster-management-hub open-cluster-management  --ignore-not-found
+             kubectl config unset contexts.wds1 --ignore-not-found
+             kubectl --context kscore delete -f https://raw.githubusercontent.com/kubestellar/kubestellar/refs/heads/release-"{{ ks_release }}"/example/kind-nginx-ingress-with-SSL-passthrough.yaml  --ignore-not-found
+      when: result is success
+
+
+    - name: install ingress controller
+      shell: |
+             kubectl config rename-context kubernetes-admin@kubernetes kscore
+             kubectl --context kscore label node --all ingress-ready="true"
+             kubectl --context kscore apply -f https://raw.githubusercontent.com/kubernetes/ingress-nginx/refs/tags/controller-v1.12.0/deploy/static/provider/kind/deploy.yaml
+
+    - name: patching nginx ingress to enable SSL passthrough
+      shell: |
+             kubectl --context kscore patch deployment ingress-nginx-controller -n ingress-nginx -p '{"spec":{"template":{"spec":{"containers":[{"name":"controller","args":["/nginx-ingress-controller","--election-id=ingress-nginx-leader","--controller-class=k8s.io/ingress-nginx","--ingress-class=nginx","--configmap=$(POD_NAMESPACE)/ingress-nginx-controller","--validating-webhook=:8443","--validating-webhook-certificate=/usr/local/certificates/cert","--validating-webhook-key=/usr/local/certificates/key","--watch-ingress-without-class=true","--publish-status-address=localhost","--enable-ssl-passthrough"]}]}}}}'
+
+    - name: extract the port number for the ingress controller
+      shell: kubectl --context kscore -n ingress-nginx get svc ingress-nginx-controller -o=jsonpath='{.spec.ports[1].nodePort}'
+      register: ingress_port
+
+    - name: install KubeStellar using core helm-cart
+      shell: |
+             export KUBESTELLAR_VERSION="{{ ks_release }}"
+             helm upgrade --install ks-core oci://ghcr.io/kubestellar/kubestellar/core-chart \
+             --version $KUBESTELLAR_VERSION \
+             --set-json='ITSes=[{"name":"its1","type":"host"}]' \
+             --set-json='WDSes=[{"name":"wds1"}]' \
+             --set-json='verbosity.default=5' \
+             --set kubeflex-operator.externalPort={{ ingress_port.stdout }}
+
+    - name: Get kubeconfig "context" for wds1 space
+      shell: |
+             kflex ctx wds1
+
+    - name: Get the token to register managed clusters
+      shell: clusteradm --context kscore get token | grep '^clusteradm join'
+      register: wec_join_command
+      until: "wec_join_command is not failed"
+      retries: 5
+      delay: 15
+             
+    -  name: write wec_join_command to file
+       ansible.builtin.copy:
+         content: "{{ wec_join_command.stdout }}"
+         dest:  /tmp/wec_join_command
+    
+    - name: Copy join command to local file (Ansible control node)
+      ansible.builtin.fetch:
+        src: /tmp/wec_join_command
+        dest: /tmp/
```

Comment:
This seems like not a great place, for a couple of reasons.

1. I find that reboot tends to clear out /tmp. Forbidding reboot seems like a bad idea. And it is not documented.
2. This does not allow me to administer multiple VPCs.

Reply:
Updated. Using now the ./data directory to store `wec_join_command` file

Reply:
Without including region and VPC name in the filename, this forbids me to administer multiple VPCs at a time.

Reply:
Fixed! using the following filename instead: `wec_join_command_{{ region }}_{{ vpc_name }}`

Reply:
I see that the `vpc_helper_vars.yaml` file puts the region and vpc qualifiers at the start rather than the end. I think that consistency would be good (but will not insist on it). Putting the qualifiers at the start gives better organized listing.

Reply:
ok, I updated with the following file name instead: `{{ region }}_{{ vpc_name }}_wec_join_command`

Reply:
I still see some unqualified filenames, testing commit 4c8ed2aa507db13c4e25ba505abb0842313145d8.

```console
bash-3.2$ ls -l .data
total 80
-rw-r-----@ 1 mspreitz  staff   5656 Mar 14 00:52 admin.conf
-rw-r--r--  1 mspreitz  staff  18043 Mar 14 00:49 ec2_instances_core.json
-rw-r--r--  1 mspreitz  staff    147 Mar 14 00:49 hosts_core
-rw-r--r--  1 mspreitz  staff    252 Mar 14 00:48 us-east-2_mspreitz_vpc_helper_vars.yaml
-rw-r--r--  1 mspreitz  staff   1101 Mar 14 00:58 us-east-2_mspreitz_wec_join_command
```

Reply:
```console
bash-3.2$ git rev-parse HEAD
b00fdae41b4883c9af6c171759c855153fb6fe08

bash-3.2$ ls -l .data
total 120
-rw-r-----@ 1 mspreitz  staff  11545 Mar 14 02:20 admin.conf
-rw-r--r--  1 mspreitz  staff  18031 Mar 14 01:58 ec2_instances_core.json
-rw-r--r--  1 mspreitz  staff  12056 Mar 14 02:24 ec2_instances_wec.json
-rw-r--r--  1 mspreitz  staff    147 Mar 14 01:58 hosts_core
-rw-r--r--@ 1 mspreitz  staff    147 Mar 14 02:25 hosts_wec
-rw-r--r--  1 mspreitz  staff    252 Mar 14 01:55 us-east-2_mspreitz_vpc_helper_vars.yaml
-rw-r--r--  1 mspreitz  staff   1100 Mar 14 02:08 us-east-2_mspreitz_wec_join_command
drwxr-xr-x  4 mspreitz  staff    128 Mar 14 02:28 wecs_config
```

Reply:
Fixed. I updated the code that there is a directory created named `<region>_<vpc_name>` that includes all the data created for the specific `vpc_name` and `region`: 

```console
tree .data/
.data/
└── us-east-1_braulio-test
    ├── admin.conf
    ├── ec2_instances_core.json
    ├── ec2_instances_wec.json
    ├── hosts_core
    ├── hosts_wec
    ├── us-east-1_braulio-test_wec_join_command
    ├── vpc_helper_vars.yaml
    └── wecs_config
        ├── 44-202-49-211.conf
        └── 44-203-174-106.conf
```